libcxx/test/std/atomics/atomics.ref/address.pass.cpp

```diff
@@ -0,0 +1,37 @@
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: std-at-least-c++26
+
+// XFAIL: !has-64-bit-atomics
+// XFAIL: !has-1024-bit-atomics
+
+// constexpr T* address() const noexcept;
+
+#include <atomic>
+#include <cassert>
+
+#include "atomic_helpers.h"
+#include "test_macros.h"
+
+template <typename T>
+struct TestAddress {
+  void operator()() const {
+    T x(T(1));
+    std::atomic_ref<T> const a(x);
+
+    assert(a.address() == std::addressof(x));
+
+    static_assert(noexcept((a.address())));
```

Comment:
`constexpr` on `address` doesn't seem tested. I think it's better to reorganize `TestAddress` to test `constexpr` together.